src/base/bittorrent/magneturi.cpp

```diff
@@ -83,10 +83,10 @@ MagnetUri::MagnetUri(const QString &source)
     m_hash = m_addTorrentParams.info_hash;
     m_name = QString::fromStdString(m_addTorrentParams.name);
 
-    foreach (const std::string &tracker, m_addTorrentParams.trackers)
+    for (const std::string &tracker : m_addTorrentParams.trackers)
         m_trackers.append(QString::fromStdString(tracker));
 
-    foreach (const std::string &urlSeed, m_addTorrentParams.url_seeds)
+    for (const std::string &urlSeed : m_addTorrentParams.url_seeds)
```

Comment:
seems need to add `qAsConst` for both loops.

Reply:
But these are `std::vector` containers and this guy says:
>Avoiding Detaching
>
>If your container is a std:: container or QVarLengthArray, you replace the `Q_FOREACH(a, b)` with `for (a : b)` and you are done. Arguably, Q_FOREACH should never, ever have been used on such a container, since copying those always copies all elements (deep copy).

https://www.kdab.com/goodbye-q_foreach/

What am i missing?

Reply:
I didn't realize they are not Qt containers, so it's OK.